dataprofiler/profilers/profile_builder.py

```diff
@@ -1431,53 +1436,61 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
-    def _get_correlation(self, clean_samples):
+    def _get_correlation_and_properties(self, data):
         """
-        Calculate correlation matrix on the cleaned data.
+        Calculate correlation matrix and necessary batch properties on the data.
 
-        :param clean_samples: the input cleaned dataset
-        :type clean_samples: dict()
+        :param data: the input dataset
+        :type data: pd.Dataframe
         """
         columns = self.options.correlation.columns
         clean_column_ids = []
         if columns is None:
             for idx in range(len(self._profile)):
-                if (self._profile[idx].profile['data_type'] not in ['int', 'float']
-                        or self._profile[idx].null_count > 0):
-                    clean_samples.pop(idx)
-                else:
+                if self._profile[idx].profile['data_type'] in ['int', 'float']:
                     clean_column_ids.append(idx)
 
-        data = pd.DataFrame(clean_samples)
-        data = data.apply(pd.to_numeric, errors='coerce')
+        # Exclude all rows with NaN in numeric columns
+        # and add dropped-NaN stats to properties for
+        # future updating
+        data_dropped = data.dropna(subset=data.columns[clean_column_ids])
+        properties = {
+            'mean': np.full(len(self._profile), np.nan),
+            'std': np.full(len(self._profile), np.nan)
+        }
+        for id in clean_column_ids:
+            properties['mean'][id] = data_dropped.iloc[:, id].mean()
+            properties['std'][id] = data_dropped.iloc[:, id].std()
+        properties['count'] = len(data_dropped.index)
```

Comment:
Instead of recalculating mean / std we should utilize the pre-calculated values and do an update to them specifically for these alterations due to NaNs. We want to reduce as many repeat calculations as possible. 

Additionally, we need to decide whether the NaNs apply across the dataset or within col to col interactions.

Reply:
The concern I saw with updating the given values is that the values that need to be removed from the mean calculation are dependent on the other columns containing NaN values. Searching for which values need to be removed and then updating the mean seemed like an equally expensive operation to me. 

In regards to applying NaN removal to the whole dataset vs pairwise, it becomes quite complex of a process to update correlations pairwise with NaNs, because then each column could potentially have different "means/stddevs" depending on what indices are NaN in one column vs another, and would require needing to find a way to store multiple different values for means and standard deviations for different columns. Thoughts? @JGSweets